example/src/locale.rs

```diff
@@ -10,6 +10,7 @@ pub mod ffi {
     impl ICU4XLocale {
         /// Construct an [`ICU4XLocale`] from a locale identifier represented as a string.
         #[diplomat::attr(supports = constructors, constructor)]
+        #[diplomat::demo(input(name(label = "Locale Name")))]
         pub fn new(name: &DiplomatStr) -> Box<ICU4XLocale> {
```

Comment:
Suggestion: `name` could be changed to `locale_name`. This improvement could be applied elsewhere, too.

Reply:
I don't think that's an improvement, having a `locale_` prefix in `Locale::new` seems redundant.

Reply:
It's redundant, but that's not a bad thing

1. In the WASM demo, the parameter name loses context when it is pulled into terminus functions
2. Code completion tools like to use parameter names as local variable names, and a local variable should be named `locale_name`, not `name`

For the WASM demo, there are a few approaches we could take

1. Annotate every function argument with its WASM Demo label (I am strongly opposed to this: a key part of the project for me has been "the WASM Demo generator should 'just work' and requiring annotations for sensible output is the opposite of that)
2. Proceed as I proposed here, by improving parameter names to functions and making them more context-free
3. A third approach if you really don't like the redundancy would be for the demo generator to remember the class name and glue them together when making labels. The label for the parameter in `ICU4XLocale::new(name)` would become "ICU4X Locale Name" (or "Locale Name" after the "ICU4X" prefix is removed in favor of namespaces).

I still think making the parameter names context-free is the better overall approach, because it is more clear / removes ambiguity and because it is a way to directly express the intent of the parameter name rather than relying on black box machinery.  2 > 3 >> 1

Reply:
It should definitely be option 3. Otherwise we end up with things like 

```rust
pub fn create_from_iso_in_calendar(
            datetime_year: i32,
            datetime_month: u8,
            datetime_day: u8,
            datetime_hour: u8,
            datetime_minute: u8,
            datetime_second: u8,
            datetime_nanosecond: u32,
            datetime_calendar: &Calendar,
        ) -> Result<Box<DateTime>, CalendarError>
```

Reply:
No, the term `year` is sufficiently context-free that it doesn't really need a prefix (technically it should be called `iso_year` though). It's only terms like `name` and `id` that really need context. This is what option 2 lets us do.

Reply:
You could have a `Date` and a `DateTime`, both have `year` arguments, which by your reasoning is confusing.

Reply:
It's not ambiguous because a terminal function shouldn't usually have both `Date` and `DateTime` as arguments.